### PR TITLE
Updating redemption related parameters

### PR DESCRIPTION
**Pull request by tomaszslabon**
Refs #https://github.com/keep-network/tbtc-v2/issues/152
This PR adds functionalities for updating parameters related to redemptions in Bridge.

### CODE REVIEW

**Comment by pdyraga** [comment]
👀

File: solidity/test/bridge/Bridge.test.ts [hunk 1]
```diff
@@ -269,6 +269,108 @@ describe("Bridge", () => {
     })
   })
 
+  describe.only("updateRedemptionParameters", () => {
```
Comment (pdyraga) [inline solidity/test/bridge/Bridge.test.ts:272]: `.only` 😬

Comment (pdyraga) [reply solidity/test/bridge/Bridge.test.ts]: Removed in https://github.com/keep-network/tbtc-v2/pull/224/commits/30cfb21bcebe111cafd3696953809f19e4528e05.

**Review by pdyraga — APPROVED** [review]

**Comment by pdyraga** [comment]
Let's address the review comments in a followup PR.

File: solidity/contracts/bridge/Bridge.sol [hunk 2]
```diff
@@ -802,7 +809,51 @@ contract Bridge is Ownable, EcdsaWalletOwner {
...
+    ///        given redemption transaction. If the maximum BTC transaction fee
+    ///        is exceeded, such transaction is considered a fraud.
+    /// @param redemptionTimeout New value of the redemption timeout in seconds.
+    ///        It is the time after which the redemption request can be reported
+    ///        as timed out. It is counted from the moment when the redemption
+    ///        request was created via `requestRedemption` call. Reported  timed
+    ///        out requests are cancelled and locked TBTC is returned to the
+    ///        redeemer in full amount.
+    /// @dev Requirements:
+    ///      - Redemption treasury fee divisor must be greater than zero
```
[the remark this answers is not in the document]
Comment (lukasz-zimnoch) [reply solidity/contracts/bridge/Bridge.sol:841]: I think the DAO can just set `type(uint64).max` in that case and we don't need to introduce special treatment for `0` value.

File: solidity/test/bridge/Bridge.test.ts [hunk 3]
```diff
@@ -269,6 +269,108 @@ describe("Bridge", () => {
     })
   })
 
+  describe("updateRedemptionParameters", () => {
```
[the remark this answers is not in the document]
Comment (lukasz-zimnoch) [reply solidity/test/bridge/Bridge.test.ts:272]: There is an issue for that: https://github.com/keep-network/tbtc-v2/issues/173

File: solidity/contracts/bridge/Bridge.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (pdyraga) [reply solidity/contracts/bridge/Bridge.sol:841]: Yeah, this is simpler.

File: solidity/test/bridge/Bridge.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (tomaszslabon) [reply solidity/test/bridge/Bridge.test.ts:272]: Done in 49f3a5d7a6cabc18a383e4b72c7f4077cdf79499.

File: solidity/contracts/bridge/Bridge.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (tomaszslabon) [reply solidity/contracts/bridge/Bridge.sol:841]: Done in e9e43f56d183a1c1f1a981d97e84a34e3430f6d6.

[the remark this answers is not in the document]
Comment (tomaszslabon) [reply solidity/contracts/bridge/Bridge.sol:841]: Yes, done in 1db860a8bc807d463ad16114905d97ebbd1b6a30.

File: solidity/contracts/bridge/Bridge.sol [hunk 4]
```diff
@@ -802,7 +809,51 @@ contract Bridge is Ownable, EcdsaWalletOwner {
...
+    ///        computed as follows:
+    ///        `treasuryFee = requestedAmount / redemptionTreasuryFeeDivisor`
+    ///        For example, if the treasury fee needs to be 2% of each
+    ///        redemption request, the `redemptionTreasuryFeeDivisor` should
+    ///        be set to `50` because `1/50 = 0.02 = 2%`.
+    /// @param redemptionTxMaxFee New value of the redemption transaction max
+    ///        fee in satoshis. It is the maximum amount of BTC transaction fee
+    ///        that can be incurred by each redemption request being part of the
+    ///        given redemption transaction. If the maximum BTC transaction fee
+    ///        is exceeded, such transaction is considered a fraud.
```
[the remark this answers is not in the document]
Comment (tomaszslabon) [reply solidity/contracts/bridge/Bridge.sol:833]: Done in 5ee48ae96e87fb8dc12c06dd1f4c9f8c24e95c8f.

**Pull request merged**